server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -4172,7 +4172,12 @@ public String toString() {
         return "IndexShard(shardRouting=" + shardRouting + ")";
     }
 
+    @Deprecated
     public void waitForSegmentGeneration(long segmentGeneration, ActionListener<Long> listener) {
```

Comment:
Same here, can we explain why it is deprecated and point to the primary term aware version?